src/components/nmrfile-browser/NMRFileBrowser.tsx

```diff
@@ -0,0 +1,507 @@
+    return '';
+  }
+  const source = entry.children[0].source;
+  const relativePath = findCommonParentFolder(
+    entry.children.map((child: NMREntryChild) => child.relativePath),
+  );
+  const params = new URLSearchParams();
+  params.set('source', source);
+  params.set('relativePath', relativePath);
+  const link = `${dbURL}v1/getZip?${params.toString()}`;
+
+  return (
+    <AnchorButton
+      href={link}
+      onClick={(e) => {
+        e.stopPropagation();
+      }}
+      style={{
+        borderRadius: '5px',
+        padding: '5px',
+      }}
+      minimal
+      icon="import"
+    />
+  );
+}
+
+function getDim(child: NMREntryChild) {
+  if (child.is1D) {
+    return '1D';
+  }
+  if (child.is2D) {
+    return '2D';
+  }
+  return '';
+}
+
+function getType(child: NMREntryChild) {
+  if (child.isFid) {
+    return 'FID';
+  }
+  if (child.isFt) {
+    return 'FT';
+  }
+  return '';
+}
+
+export interface NMRFileBrowserProps {
+  setSpectra?: (ids: string | string[]) => void;
+  appendSpectra?: (ids: string | string[]) => void;
+}
+
+export function NMRFileBrowser(props: NMRFileBrowserProps) {
+  const { setSpectra, appendSpectra } = props;
+
+  const [entries, setEntries] = useState<NMREntry[]>([]);
+  const [opened, setOpened] = useState<string>('');
+  const [total, setTotal] = useState(0);
+
+  function getLink(entry: NMREntry, showAll: boolean) {
+    const labels: Record<SpectrumKinds, string> = {
+      all: 'All',
+      oneD: '1D',
+      twoD: '2D',
+      fid: 'FID',
+      ft: 'FT',
+    };
+    const icons: Record<SpectrumKinds, BlueprintIcons | null> = {
+      all: 'multi-select',
+      oneD: 'pulse',
+      twoD: 'scatter-plot',
+      fid: null,
+      ft: null,
+    };
+    const otherKinds: SpectrumKinds[] = ['oneD', 'twoD', 'fid', 'ft'];
+    const kindsRender: SpectrumKinds[] = ['all'];
+
+    if (showAll) {
+      kindsRender.unshift(...otherKinds);
+    }
+    return kindsRender.map((kind) => {
+      const label = labels[kind];
+      const icon = icons[kind];
+      const ids = entry.links[kind].ids;
+      return (
+        <Button
+          key={kind}
+          onClick={(e) => {
+            e.stopPropagation();
+            setSpectra?.(ids);
+          }}
+          tooltipProps={
+            ids.length > 0
+              ? {
+                  content: `Nb ${label} spectra: ${ids.length}`,
+                  position: 'bottom-left',
+                }
+              : {}
+          }
+          css={css({
+            borderRadius: '5px',
+            padding: '5px',
+          })}
+          disabled={ids.length === 0}
+          icon={icon}
+          minimal={ids.length > 0}
+        >
+          {icon ? null : label}
+        </Button>
+      );
+    });
+  }
+  const displayFirst = 100;
+  useEffect(() => {
+    processQuery('')
+      .then((entries) => {
+        setEntries(entries);
+        setTotal(entries.length);
+      })
+      .catch(() => {});
+  }, []);
```

Comment:
Please use `@tanstack/react-query` for queries, not manual management with useEffect